### ISSUE TITLE
Make trusted Teleporters and Reserves queryable through Runtime APIs

### ISSUE THREAD

**Issue by joepetrowski** [paritytech/polkadot-sdk#97]
Some applications would like to know if certain chains are Trusted Teleporters or not. This will affect how they treat DOT/KSM deposits on those chains, or on the extrinsics they construct to interact with the Relay Chain.

AFACT, one must look up the Trusted Teleporters in the source code, e.g. https://github.com/paritytech/polkadot/blob/release-v0.9.19/runtime/polkadot/src/xcm_config.rs#L114

**Comment by Polkadot-Forum** [comment]
This issue has been mentioned on **Polkadot Forum**. There might be relevant details there:

https://forum.polkadot.network/t/xcm-asset-registry/1168/8

**Comment by acatangiu** [comment]
Can we expose this as runtime APIs rather than adding to Metadata?

The configuration is a filter not an actual list of (asset, location).
Metadata would have to export this filter somehow and consumers can call it offchain.
Runtime APIs can take (asset, location) and provide `true`/`false` based on filter (whether `asset` can be teleported to `location`).

Any particular benefit for one over the other?

**Comment by joepetrowski** [comment]
Yeah good point, runtime API would be fine IMO. Just as long as wallets can easily get the information.

**Comment by acatangiu** [comment]
We'll prioritize this. It's easy and important.

We'll provide runtime APIs for:
- querying trusted teleporters
- querying trusted reserves

**Comment by franciscoaguirre** [comment]
We could extend this to reserves and have another API for it. They have the same format `(asset, location)`.

**Comment by x3c41a** [comment]
The issue will be closed once these API methods are integrated in fellowship-runtimes Polkadot and Kusama
